### PR TITLE
Upgrade nodejs runtime for cdk-codepipeline-slack lambdas

### PR DESCRIPTION
**Pull request by calvinwyoung**
Using the `SlackApprovalAction` construct from `@cloudcomponents/cdk-codepipeline-slack`, currently results in the following error:

```
Resource handler returned message: "The runtime parameter of nodejs10.x is no longer supported for 
creating or updating AWS Lambda functions. We recommend you use the new runtime (nodejs14.x) while
creating or updating functions.
```

This was reported in #143.

This PR bumps the nodejs runtime for the Slack notifier lambda from 10.x to 14.x.

### CODE REVIEW

File: packages/cdk-contentful-webhook/lambda-file-sizes.json [hunk 1]
```diff
@@ -1 +1 @@
...
..."contentful-webhook/index.js","previous":203440,"size":203418,"diff":-22}]},{"timestamp":1599246565057,"files":[{"filename":"contentful-webhook/index.js","previous":202993,"size":203440,"diff":447}]},{"timestamp":1598552652514,"files":[{"filename":"contentful-webhook/index.js","previous":202992,"size":202993,"diff":1}]},{"timestamp":1596542526761,"files":[{"filename":"contentful-webhook/index.js","previous":207102,"size":202992,"diff":-4110}]},{"timestamp":1596540034638,"files":[{"filename":"contentful-webhook/index.js","previous":206593,"size":207102,"diff":509}]},{"timestamp":1596457456615,"files":[{"filename":"contentful-webhook/index.js","previous":206357,"size":206593,"diff":236}]},{"timestamp":1596440551356,"files":[{"filename":"contentful-webhook/index.js","previous":206424,"size":206357,"diff":-67}]},{"timestamp":1596437759972,"files":[{"filename":"contentful-webhook/index.js","previous":541950,"size":206424,"diff":-335526}]},{"timestamp":1596413723535,"files":[{"filename":"contentful-webhook/index.js","previous":542071,"size":541950,"diff":-121}]},{"timestamp":1596407623809,"files":[{"filename":"contentful-webhook/index.js","previous":191169,"size":542071,"diff":350902}]}]
```
Comment (calvinwyoung) [inline packages/cdk-contentful-webhook/lambda-file-sizes.json:1]: These files were automatically updated after I ran `yarn test` locally. I assumed we might want to update this in version control but I'm not entirely sure. Please let me know if you'd like me to revert this.

**Review by hupe1980 — APPROVED** [review]
LGTM! Thx

**Comment by zvictor** [comment]
The problem is back:

> The runtime parameter of nodejs14.x is no longer supported for creating or updating AWS Lambda functions. We recommend you use the new runtime (nodejs20.x) while creating or updating functions. (Service: Lambda, Status Code: 400, Request ID: d24e03df-b5d1-4af2-8468-27027cc04f27)

**Comment by Prasad-Telasula** [comment]
Yes !!! 
I concur with @zvictor regarding the obsolescence of nodejs14.x support within AWS. It's imperative that we facilitate an upgrade to a supported version of Node.js. Could anyone provide an update on whether there is active development underway to address this issue?

Resource handler returned message: "The runtime parameter of nodejs14.x is no longer supported for creating or updating AWS Lambda functions. We recommend you use the new runtime (nodejs20.x) while creating or updating functions. (Service: Lambda, Status Code: 400, Request ID: d898554a-db42-44e0-8580-a68591c283e0)" (RequestToken: abe27e74-99a4-2c4d-ee02-4b514bae0f74, HandlerErrorCode: InvalidRequest)

**Comment by kadishmal** [comment]
Replied in https://github.com/cloudcomponents/cdk-constructs/issues/208#issuecomment-2111766163 that Node v18 and v20 are supported on the latest CDK v2 releases. CDK v1 based releases are no longer maintained.

**Pull request merged**